docs/migrations/migrations_guide.md

```diff
@@ -0,0 +1,192 @@
+# 📚 Database Migrations Guide – Nixopus
+
+This guide explains how database migrations work in Nixopus, how to create and apply them, and best practices for using them effectively.
+
+---
+
+## 🧰 Tool We Use: `golang-migrate`
+
+Nixopus uses [`golang-migrate`](https://github.com/golang-migrate/migrate) to manage SQL-based database migrations in Go. This tool supports programmatic and CLI-based migration execution, and we’ve embedded it into our application lifecycle so that migrations run automatically on startup.
+
+---
+
+## 📂 Where Migrations Live
+
+All migrations are stored in the `migrations/` directory (relative to the module where migration logic exists). Each migration consists of two SQL files:
+
+```
+<version>_<name>.up.sql   # Migration to apply
+<version>_<name>.down.sql # Migration to rollback
```

Comment:
Right now, we haven't added/maintained any version yet; this is purely sequential. Nothing specific to versioning with that..

```
<sequence_number>_<descriptive_resource_name}_<up|down>.sql
```